hashicorp/vault
# Add storybook

# Add Storybook
This PR implements Storybook as a living style guide to catalog components in the Vault UI.

<img width="1005" alt="Screen Shot 2019-03-29 at 4 11 41 PM" src="https://user-images.githubusercontent.com/903288/55267086-63a8cb80-523d-11e9-87bc-854b13869e8a.png">

![storybook-demo](https://user-images.githubusercontent.com/903288/55198497-75be3780-5173-11e9-9dc1-d52fb42148d6.gif)

### Features
- View live examples of Vault's components
- Change the viewport to see components as they appear on different devices
- View component documentation, example usage, and and links to source code on the 'Notes' tab
- Automatically generate story and notes files for each component

______

### Testing
- Boot up Storybook with `yarn storybook`
- Verify that all components show up in the left nav (note: you'll have to refresh upon initial page load. this is a known bug with Storybook Ember).
- Verify that each component has at least 1 example in the canvas tab
- Verify that each component notes tab has a description, links to source code, a table of properties, and an example of how to use the component
- Verify that you can generate a new story file `ember generate story <component-name>`
- Verify that you can generate the component notes markdown file `yarn gen-story-md [name-of-component]`

### Feedback
- Does the file structure and organization of components in the left nav make sense?
- Are the examples we've shown in the canvas tab helpful?
- Are there any components which should use knobs (ways to dynamically change a components properties) that don't?
- Do the component descriptions and properties table make sense?

Previous discussion related to this PR is on https://github.com/hashicorp/vault/pull/6375
